python/lsst/resources/_resourcePath.py

```diff
@@ -602,15 +616,22 @@ def getExtension(self) -> str:
             indicating file compression, in which case the combined
             extension (e.g. ``.fits.gz``) will be returned.
         """
+        if self.isdir():
+            return ""
+
```

Comment:
For schemeless and file forceDirectory=None, this now calls `stat` where previously no I/O was necessary.

Reply:
This one made no sense at all so it now returns the extension regardless of whether it's a directory or not.